packages/headless/doc-parser/doc-parser.ts

```diff
@@ -98,3 +105,48 @@ function resolveUseCase(useCase: UseCase): ResolvedUseCase {
 const resolved = useCases.map(resolveUseCase);
 
 writeFileSync('dist/parsed_doc.json', JSON.stringify(resolved, null, 2));
+
+interface SSRUseCaseConfiguration {
+  controllers: SSRControllerConfiguration[];
+  engine: EngineConfiguration;
+}
+
+interface SSRUseCase {
+  name: string;
+  entryFile: string;
+  config: SSRUseCaseConfiguration;
+}
+
+interface ResolvedSSRUseCase {
+  name: string;
+  controllers: SSRController[];
+  engine: {initializer: string};
+}
+
+const ssrUseCases: SSRUseCase[] = [
+  {
+    name: 'ssr.search',
+    entryFile: 'temp/ssr.search.api.json',
+    config: ssrSearchUseCase,
+  },
+];
+
+function resolveSSRUseCase(ssrUseCase: SSRUseCase): ResolvedSSRUseCase {
+  const {name, entryFile, config} = ssrUseCase;
+  process.env['currentUseCaseName'] = name;
+  const apiModel = new ApiModel();
+  const apiPackage = apiModel.loadPackage(entryFile);
+  const entryPoint = apiPackage.entryPoints[0];
+
+  const controllers = config.controllers.map((controller) =>
+    resolveSSRController(entryPoint, controller.initializer)
+  );
+
+  const engine = {initializer: config.engine.initializer};
+
+  return {name, controllers, engine};
+}
+
+const ssrResolved = ssrUseCases.map(resolveSSRUseCase);
+
+writeFileSync('dist/ssr_parsed_doc.json', JSON.stringify(ssrResolved, null, 2));
```

Comment:
This is copying the same thing that the doc parser already does for standard use case but puts it in a new ssr_parsed_doc.json. The reason they are separated is that they are very different, one of them has a 'Function' initializer and the other is a 'Variable'. They either had to be added to the current json files or separated. I decided to separate them.